### ISSUE TITLE
tests in `search`

### ISSUE THREAD

**Issue by vil02** [TheAlgorithms/Go#575]
The [_search functions_](https://github.com/TheAlgorithms/Go/tree/master/search) cannot be treated _uniformly_ in tests. The issue arises when the input array contains duplicated entries.

Consider the example when we want to fund the index of `4` in the array `[1, 2, 4, 4, 4, 6, 7, 8, 9, 10]` (note that the array is sorted):
- [`Binary`](https://github.com/TheAlgorithms/Go/blob/d61cee077bf76f9bc5c16ee4eb88385d1796b03c/search/binary.go#L6) and [`BinaryIterative`](https://github.com/TheAlgorithms/Go/blob/d61cee077bf76f9bc5c16ee4eb88385d1796b03c/search/binary.go#L23) will return 4,
- [`Jump2`](https://github.com/TheAlgorithms/Go/blob/d61cee077bf76f9bc5c16ee4eb88385d1796b03c/search/jump2.go#L5) will return 4,
- [`Linear`](https://github.com/TheAlgorithms/Go/blob/d61cee077bf76f9bc5c16ee4eb88385d1796b03c/search/linear.go#L4) and [`Interpolation`](https://github.com/TheAlgorithms/Go/blob/d61cee077bf76f9bc5c16ee4eb88385d1796b03c/search/interpolation.go#L15) will return 2.
- [`Jump`](https://github.com/TheAlgorithms/Go/blob/master/search/jump.go#L15) will return 2.

To summarize the functions like [`Linear`](https://github.com/TheAlgorithms/Go/blob/d61cee077bf76f9bc5c16ee4eb88385d1796b03c/search/linear.go#L4) or [`Interpolation`](https://github.com/TheAlgorithms/Go/blob/d61cee077bf76f9bc5c16ee4eb88385d1796b03c/search/interpolation.go#L15) return the smallest index of the searched element (if it exists in the array). While other functions like [`Binary`](https://github.com/TheAlgorithms/Go/blob/d61cee077bf76f9bc5c16ee4eb88385d1796b03c/search/binary.go#L6) or [`Jump2`](https://github.com/TheAlgorithms/Go/blob/d61cee077bf76f9bc5c16ee4eb88385d1796b03c/search/jump2.go#L5) return _some_ index.

I think the above shows a need to change tests of the `search` package.

**Comment by Nd1097** [comment]
Hello, would like to contribute to this. Can you assign the task?

**Comment by siriak** [comment]
I agree with `for example the Interpolation function could be simplified to find some index`. @vil02 do you want to implement this yourself?

**Comment by vil02** [comment]
> I agree with `for example the Interpolation function could be simplified to find some index`. @vil02 do you want to implement this yourself?

I think @Nd1097 wanted to do it. If there will be no other volunteer you can assign me.

**Comment by KreanXie** [comment]
I don`t think the [Binary](https://github.com/TheAlgorithms/Go/blob/d61cee077bf76f9bc5c16ee4eb88385d1796b03c/search/binary.go#L6) func is useful when the array has repeated elements, maybe we can just refer to the sort.Search func in the standard library, which has a more clear and strict definition.

**Comment by aminoxix** [comment]
if the binary func intends to find the smallest index, you might be referring to lower & upper bound algorithm.

ref.: https://peng.fyi/post/binary-search-in-go-sdk

we can segregate the concepts to attend this issue.

also, agreed with @tjgurwara99

**Comment by aminoxix** [comment]
btw, any update? @Nd1097
